### PR TITLE
Implement up command

### PR DESCRIPTION
**Pull request by rtluckie**
- implement up cmd
- quiet output of user provision

### CODE REVIEW

File: files/provision-user [hunk 1]
```diff
@@ -5,7 +5,7 @@ OS=$(grep "^ID" /etc/os-release | cut -d= -f2)  # debian | centos
 SUDO_GROUP=$([ "$OS" == "debian" ] && echo "sudo" || echo "wheel")
 
 if [ ! -f "/etc/kdk/provisioned" ]; then
-    useradd ${KDK_USERNAME} -m -G ${SUDO_GROUP},docker -s ${KDK_SHELL}
+    useradd ${KDK_USERNAME} -m -G ${SUDO_GROUP},docker -s ${KDK_SHELL} > /dev/null
```
Comment (dcwangmit01) [inline files/provision-user]: Why is this necessary now but not before?

File: cmd/up.go [hunk 2]
```diff
@@ -15,17 +15,80 @@
...
+					fmt.Sprintf("KDK_USERNAME=%s", viper.Get("docker.environment.KDK_USERNAME").(string)),
+					fmt.Sprintf("KDK_SHELL=%s", viper.Get("docker.environment.KDK_SHELL").(string)),
+					fmt.Sprintf("KDK_DOTFILES_REPO=%s", viper.Get("docker.environment.KDK_DOTFILES_REPO").(string)),
+				},
+				ExposedPorts: nat.PortSet{
+					"2022": struct{}{},
+				},
+			},
+			&container.HostConfig{
+				Privileged: true,
```
Comment (dcwangmit01) [inline cmd/up.go]: I don't think privileged was required before.  Why now?  Bind mounts?

File: files/provision-user (hunk 1 above)
Comment (rtluckie) [reply files/provision-user]: not necessary.  it just cleans up output.

**Comment by drekle** [comment]
I am currently running this branch successfully.

File: cmd/pull.go [hunk 3]
```diff
@@ -32,20 +34,22 @@ var pullCmd = &cobra.Command{
...
 
-		imageCoordinates := []string{viper.Get("image.repository").(string), viper.Get("image.tag").(string)}
-
 		client, err := client.NewEnvClient()
+
 		if err != nil {
 			logger.WithField("error", err).Fatal("Failed to create docker client")
 		}
 
+		imageCoordinates := strings.Join([]string{viper.Get("image.repository").(string), viper.Get("image.tag").(string)}, ":")
```
Comment (drekle) [inline cmd/pull.go]: seems a bit more complicated than...

```
imageCoordinates := fmt.Sprintf("%s:%s", viper.Get("image.repository").(string), viper.Get("image.tag").(string))
```

maybe preference

File: cmd/up.go [hunk 4]
```diff
@@ -15,17 +15,81 @@
...
+		if err != nil {
+			logger.WithField("error", err).Fatal("Failed to create docker client")
+		}
+
+		ctx := context.Background()
+
+		var binds []string
+
+		for _, bind := range viper.Get("docker.binds").([]interface{}) {
+			configBind := make(map[string]string)
```
Comment (drekle) [inline cmd/up.go]: the hard cast to string already assumes the key is a string so this map is maybe not needed.
I believe the body of this for could be one line.
https://play.golang.org/p/TsdEnhaiiX1

File: cmd/up.go [hunk 5]
```diff
@@ -15,17 +15,81 @@
...
+				configBind[key.(string)] = value.(string)
+			}
+			binds = append(binds, fmt.Sprintf("%s:%s", configBind["source"], configBind["target"]))
+		}
+
+		containerCreateResp, err := client.ContainerCreate(
+			ctx,
+			&container.Config{
+				Hostname: viper.Get("docker.hostname").(string),
+				Image:    strings.Join(imageCoordinates, ":"),
```
Comment (drekle) [inline cmd/up.go]: There is a mixture of strings.Join with fmt.Sprintf in this structure initialization.
